src/invariant/test/LiabilitiesMatchOffersTests.cpp

```diff
@@ -226,9 +222,14 @@ generateBuyingLiabilities(LedgerEntry offer, bool excess, uint32_t authorized)
     {
         auto account = LedgerTestUtils::generateValidAccountEntry();
         account.accountID = oe.sellerID;
+
+        auto minBalance = getMinBalance(app, account);
+
         auto maxBalance = INT64_MAX - oe.amount;
-        account.balance = excess ? std::max(account.balance, maxBalance + 1)
-                                 : std::min(account.balance, maxBalance);
+        account.balance =
+            excess
+                ? std::max(account.balance, maxBalance + 1)
+                : std::min(std::max(account.balance, minBalance), maxBalance);
```

Comment:
A few questions/comments:
- An account can be below it's minimum balance (it must have 0 native selling liabilities in that case). Do we actually need this change?
- You aren't ensuring that `maxBalance >= minBalance`, so clamping might not do what you expect.

Reply:
1. The line below will get hit if there are any liability changes, which is why the test case was failing when the account was below it's minimum balance. https://github.com/stellar/stellar-core/blob/9f3cc04e6ec02c38974c42545a86cdc79809252b/src/invariant/LiabilitiesMatchOffers.cpp#L256

2. This is a good point. It's possible that the generated minimum balance is so high that the test will fail unless we modify `numSponsoring`. This is very unlikely at the moment since `maxBalance` is usually very close to `INT64_MAX`, and `numSponsoring` is a `uint32`. Does this make sense and do you think it's worth addressing?